packages/patrol_cli/lib/src/crossplatform/app_options.dart

```diff
@@ -109,9 +111,21 @@ class AndroidAppOptions {
 
       cmd.add('-Pdart-defines=$dartDefinesString');
     }
+    if (flutter.dartDefineFromFile.isNotEmpty) {
+      cmd.addAll(_parseJsonToGradleConfig(flutter.dartDefineFromFile));
+    }
```

Comment:
I think that `dartDefines` should take precedence over `dartDefinesFromFile` here.

It'd be useful to write a test for this.

Reply:
do you mean to verify if the key wasn't passed in dartDefines and if it was, to take it out from dartDefinesFromFile?

Reply:
I mean something like this:

1. `EXAMPLE_KEY=original` is defined in the JSON file
2. `--dart-define EXAMPLE_KEY=overridden` is passed to a command

Expected result: `EXAMPLE_KEY=overridden` is present in the app.

Reply:
makes sense!

Reply:
so if the key exists in dart-defines and in the JSON file, override the values from the JSON file

Reply:
yup